### DISCUSSION TITLE
[SOLVED] Add-on: 'Customize JSON output'

### DISCUSSION THREAD

**Discussion by joanesinurrietasmartpm** [WeblateOrg/weblate#9291] [Q&A] [answered]
Hello,

I added some JSON with key/values for different languages.

I want them to be alphabetically sorted when I download the generated files.

I am using the standard JSON config in Weblate.

I read the documentation and I added this add-on. However, it doesn't change anything. Do I have to do something else?

![image](https://github.com/WeblateOrg/weblate/assets/131172312/01452e47-d5c5-4b51-afa9-07ee59684990)

When I click on download (As a zip or each language one by one) the file remains the same as before the add-on was installed.

Thank you in advance,

**Comment by nijel** [comment]
The add-on influences how Weblate saves the file. If it did not yet save it, it won't change it.

**Comment by joanesinurrietasmartpm** [comment]
I deleted the language within the component. Then created another instance of that language into the component and I uploaded a file already sorted. When I click download, the .JSON created is the older one without sorting. 

I also tried creating another component with the same languages and directly upload a sorted JSON file. I can't upload the same JSON file into the second component. I'm assuming it detects the same strings as the first component and weblate doesn't allow it.

In any case, now I have sorted .JSON files (Sorted manually). How can I upload them to save as they are on weblate? (To have sorted downloadable files)

**Reply by nijel** [reply]
Maybe using a different upload method will help? See https://docs.weblate.org/en/latest/user/files.html#import-methods

**Reply by joanesinurrietasmartpm** [reply]
I was missing the correct upload method because I was trying it in a language that wasn't the main one. If I choose English, this option is available:

![image](https://github.com/WeblateOrg/weblate/assets/131172312/c97d7554-98d8-4807-a139-0bbf5b311bc1)

Now, I have the correct order in the second component.

However, why does weblate maintain the original sorting when I delete a language and create a new instance uploading another file with other sorting, I don't know.

I will continue with the second component. Thank you.

**Reply by joanesinurrietasmartpm** [reply, marked as answer]
The solution for the ordering in the first component was to add another new string, after enabling the JSON add-onn with the Key sorting activated. This insert sorted the remaining languages to the alphabetical order.
